File: QuSoC/QuSoC/apps/Fibonacci/firmware/Fibonacci.cpp
#include "Fibonacci.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace Fibonacci
{
	uint32_t Firmware::Fib(uint32_t value)
	{
		switch (value)
		{
			case 0:
			{
				return 0;
			}
			case 1:
			{
				return 1;
			}
			default:
			{
				return (Fib((value - 1)) + Fib((value - 2)));
			}
		}
	}
	void Firmware::EntryPoint()
	{
		Fibonacci_SOC_Counter = Fib(Fibonacci_SOC_Counter);
	}
}

File: QuSoC/QuSoC/apps/Fibonacci/source/Fibonacci.cs
namespace Fibonacci
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
    }

    public static class Firmware
    {
        static uint Fib(uint value)
        {
            switch (value)
            {
                case 0: return 0;
                case 1: return 1;
                default: return Fib(value - 1) + Fib(value - 2);
            }
        }

        public static void EntryPoint()
        {
            SOC.Instance.Counter = Fib(SOC.Instance.Counter);
        }
    }
}
